site/docs/tasks/control-plane-security.md

```diff
@@ -12,7 +12,83 @@ The `NetworkPolicy` is always created. However, your Kubernetes cluster must hav
 
 ## Mutual TLS
 
-If Istio is installed in the cluster then you can use [automatic mutual TLS](https://istio.io/latest/docs/tasks/security/authentication/authn-policy/#auto-mutual-tls) to encrypt the traffic to the connectors.
+### Using Fybrik
+
+Fybrik can be configured to have the traffic between the manager and the connectors encrypted by using TLS. In addition, mutual TLS authentication is possible too.
+ 
+In the TLS mode, the connectors (aka the servers) should have their certificates available to provide them to the manager (aka client) in the TLS protocol handshake process. In mutual TLS mode, both the manager and connector should have their certificates available.
+
+
+#### Adding TLS Secrets
+
+The manager/connectors certificates are kept in Kubernetes secret:
+
+For each component copy its certificate into a file names tls.crt. Copy the certificate key into a file named tls.key.
+
+Use kubectl with the tls secret type to create the secrets.
+
+```bash
+kubectl -n fybrik-system create secret tls tls-manager-certs \
+  --cert=tls.crt \
+  --key=tls.key
+```
+
+[Cert-manager](https://cert-manager.io/) can also be used to generate the secret above using its [`Certificate`](https://cert-manager.io/docs/concepts/certificate/) resource. For example, the following is an example of a `Certificate` resource for the manager where a tls type secret named `tls-manager-certs` is created by the cert-manager. The `issuerRef` field points to a cert-manager resource names [`Issuer`](https://cert-manager.io/docs/configuration/ca/) that holds the information about the CA that signs the certificate.
+
+```bash
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: tls-manager-cert
+  namespace: fybrik-system
+spec:
+  dnsNames:
+  - manager
+  issuerRef:
+    kind: Issuer
+    name: ca-issuer
+  secretName: tls-manager-certs
+```
+
```

Comment:
The advantahe of using certificate manager that it automaticalliy tries to renew certificates.